File: src/main/java/com/hoi4utils/clausewitz/data/units/OrdersOfBattle.java
package com.hoi4utils.clausewitz.data.units;

import com.hoi4utils.clausewitz.HOIIVUtilsInitializer;
import com.hoi4utils.clausewitz.data.country.Character;
import com.hoi4utils.clausewitz.data.country.CountryFlag;
import com.hoi4utils.clausewitz.data.idea.Idea;
import com.hoi4utils.clausewitz.data.technology.Technology;

import java.util.Set;

// todo convert to scala code
/**
 * Refer to <url> <a href="https://hoi4.paradoxwikis.com/Division_modding">HOI4 Division Modding Wiki</a> </url>
 * todo
 */
public class OrdersOfBattle {
	// private Set<DivisionTemplate> divisionTemplates
	// set of division templates defined in oob
	// Set<effect> instantEffects
	// effect/instant effect?  (when oob loaded do this, like add eq. prod.)

    /* mostly just load the oob and its date */

    String oob_id;
    private int capital;
    private int defaultResearchSlots;               // default research slots
    private Set<CountryFlag> countryFlags;
    private double stability;                       // stability percentage defined from 0.0-1.0
    private double warSupport;                      // war support percentage defined from 0.0-1.0
    private Set<Technology> startingTech;           // starting technology defined in history/countries file
    private final int convoys = 0;
    private Set<Idea> ideas;
    private Politics politics;
    private Popularities popularities;
    private Set<Character> recruitCharacters;
    private Character countryLeader; // todo ?
}

File: src/main/scala/com/hoi4utils/clausewitz/data/units/OrdersOfBattle.scala
package com.hoi4utils.clausewitz.data.units

import com.hoi4utils.clausewitz.HOIIVUtilsInitializer
import com.hoi4utils.clausewitz.data.country.Character
import com.hoi4utils.clausewitz.data.country.CountryFlag
import com.hoi4utils.clausewitz.data.idea.Idea
import com.hoi4utils.clausewitz.data.technology.Technology
import com.hoi4utils.clausewitz.script.StringPDX

import java.util
import java.util.Set


// todo convert to scala code// todo convert to scala code

/**
 * Refer to <url> <a href="https://hoi4.paradoxwikis.com/Division_modding">HOI4 Division Modding Wiki</a> </url>
 * todo
 */
class OrdersOfBattle {
  // private Set<DivisionTemplate> divisionTemplates
  // set of division templates defined in oob
  // Set<effect> instantEffects
  // effect/instant effect?  (when oob loaded do this, like add eq. prod.)
  /* mostly just load the oob and its date */
  final val id = new StringPDX("id")
  private val capital = 0
  private val defaultResearchSlots = 0 // default research slots

  private val countryFlags: util.Set[CountryFlag] = null
  private val stability = .0 // stability percentage defined from 0.0-1.0

  private val warSupport = .0 // war support percentage defined from 0.0-1.0

  private val startingTech: util.Set[Technology] = null // starting technology defined in history/countries file

  final private val convoys = 0
//  private val ideas: util.Set[Idea] = null
  private val politics: Politics = null
  private val popularities: Popularities = null
  private val recruitCharacters: util.Set[Character] = null
  private val countryLeader: Character = null // todo ?

}

object OrdersOfBattle {
  def list: List[OrdersOfBattle] = {
    List[OrdersOfBattle]()
  }
}
